src/cmfd_data.F90

```diff
@@ -265,39 +265,39 @@ subroutine compute_xs()
 
                 ! Left surface
                 cmfd % current(1,h,i,j,k) = t % results(RESULT_SUM, 1, &
-                     score_index + OUT_LEFT)
+                     score_index + ng*OUT_LEFT)
```

Comment:
At this point, it looks to me like `score_index + 1` would correspond to `OUT_LEFT` for the given mesh cell. This means that `score_index + 1 + ng` would correspond to `IN_LEFT`, `score_index + 1 + 2*ng` would correspond to `OUT_RIGHT`, etc. Can you check my logic? If it's right, that means each expression should look something like `score_index + 1 + ng*(z - 1)` where `z` is `OUT_LEFT`, `IN_LEFT`, `OUT_RIGHT`, etc.

Reply:
You're right, my first implementation was specific to the 2 group case but didn't work for anything beyond 2 groups.